crates/bevy_text/src/lib.rs

```diff
@@ -34,35 +34,42 @@ use bevy_render::{camera::CameraUpdateSystem, ExtractSchedule, RenderApp};
 use bevy_sprite::SpriteSystem;
 use std::num::NonZeroUsize;
 
+/// Adds text rendering support to an app.
+///
+/// When the `bevy_text` feature is enabled with the `bevy` crate, this
+/// plugin is included by default in the `DefaultPlugins`.
 #[derive(Default)]
 pub struct TextPlugin;
 
-/// [`TextPlugin`] settings
+/// Settings used to configure the [`TextPlugin`].
 #[derive(Resource)]
 pub struct TextSettings {
-    /// Maximum number of font atlases supported in a [`FontAtlasSet`].
-    pub max_font_atlases: NonZeroUsize,
-    /// Allows font size to be set dynamically exceeding the amount set in `max_font_atlases`.
+    /// Soft maximum number of font atlases supported in a [`FontAtlasSet`]. When this is exceeded,
+    /// the [`FontAtlasWarning`] resource's `warning` field will be set to `true`.
+    pub soft_max_font_atlases: NonZeroUsize,
+    /// Allows font size to be set dynamically exceeding the amount set in `soft_max_font_atlases`.
     /// Note each font size has to be generated which can have a strong performance impact.
     pub allow_dynamic_font_size: bool,
 }
 
 impl Default for TextSettings {
     fn default() -> Self {
         Self {
-            max_font_atlases: NonZeroUsize::new(16).unwrap(),
+            soft_max_font_atlases: NonZeroUsize::new(16).unwrap(),
             allow_dynamic_font_size: false,
         }
     }
 }
 
+/// When the number of font atlases exceeds the [`TextSettings::soft_max_font_atlases`] setting,
+/// this resource's `warning` field will be set to `true` and a warning will be emitted.
```

Comment:
the field is called `warned` and is not public, so this is not useful as documentation